### PR TITLE
test: add transport tests

### PR DESCRIPTION
**Pull request by nechaido**

### CODE REVIEW

File: test/node/transport-tls.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            `api.${iface} must include ${method}`)
+        );
+      });
+
+      test.end();
+    }
+  );
+});
+
+const freePort = 65535;
```
Comment (aqrln) [inline test/node/transport-tls.js]: There's no guarantee this port will be free. Maybe use an IP from  240.0.0.0/4 instead?

File: test/node/transport-net-tcp.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+test.test('TCP connection must connect and inspect', (test) => {
+  jstp.net.connectAndInspect(app.name, null, interfaces, port,
+    (error, conn, api) => {
+      connection = conn;
+      test.assertNot(error, 'connectAndInspect must not return an error');
+
+      interfaces.forEach((iface) => {
+        test.assert(iface in api, `api must include '${iface}'`);
+        Object.keys(app.interfaces[iface]).forEach(method =>
+          test.assert(method in api[iface],
```
Comment (belochub) [inline test/node/transport-net-tcp.js]: You should use braces with the lambda in this case.

File: test/node/transport-tls.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+test.test('TLS connection must connect and inspect', (test) => {
+  jstp.tls.connectAndInspect(app.name, null, interfaces, port,
+    (error, conn, api) => {
+      connection = conn;
+      test.assertNot(error, 'connectAndInspect must not return an error');
+
+      interfaces.forEach((iface) => {
+        test.assert(iface in api, `api must include '${iface}'`);
+        Object.keys(app.interfaces[iface]).forEach(method =>
+          test.assert(method in api[iface],
```
Comment (belochub) [inline test/node/transport-tls.js]: Ditto.

File: test/node/transport-tls.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+const app = require('../fixtures/application');
+const application = new jstp.Application(app.name, app.interfaces);
+
+const interfaces = Object.keys(app.interfaces);
+
+// Prevent Node.js from throwing DEPTH_ZERO_SELF_SIGNED_CERT.
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const key = fs.readFileSync(
+  path.resolve(__dirname, '../fixtures/cert/test.key'));
```
Comment (belochub) [inline test/node/transport-tls.js]: Shouldn't the closing parenthesis be placed on the next line here?

File: test/node/transport-net-ipc.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+test.test('IPC connection must connect to server', (test) => {
+  jstp.net.connect(app.name, null, socket, (error, conn) => {
+    connection = conn;
+    test.assertNot(error, 'connect must not return an error');
+    test.end();
+  });
+});
+
+test.test('IPC connection must connect and inspect', (test) => {
+  jstp.net.connectAndInspect(app.name, null, interfaces, socket,
```
Comment (aqrln) [inline test/node/transport-net-ipc.js]: Nit: maybe place these arguments on a new line.

File: test/node/transport-net-ipc.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    (error, conn, api) => {
+      connection = conn;
+      test.assertNot(error, 'connectAndInspect must not return an error');
+
+      interfaces.forEach((iface) => {
+        test.assert(iface in api, `api must include '${iface}'`);
+        Object.keys(app.interfaces[iface]).forEach((method) => {
+          test.assert(method in api[iface],
```
Comment (aqrln) [inline test/node/transport-net-ipc.js]: Nit: I think we have agreed on style like this:

```js
test.assert(
  method in api[iface],
  `...`
);
```

File: test/node/transport-net-ipc.js (hunk 5 above)
Comment (aqrln) [inline test/node/transport-net-ipc.js]: This nit and the other one apply to all the files in this PR.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/jstp/commit/23c3a6bff233c21d9b7198efa63ec31194089440.

**Pull request closed without merging**